backend/test/modules/player-profile/core/application/SendEmailAfterProfileWasCreated.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { testPlayerProfileModule } from './TestPlayerProfileModule';
+import { PlayerProfileWasCreated } from '../../../../../src/modules/player-profiles/core/domain/event/PlayerProfileWasCreated';
+import {CommandPublisherMock} from "../../../../test-support/shared/core/CommandPublisherMock";
+
+describe('Send an email to player', () => {
+  it('When player profile is created, then execute command for email sending', async () => {
+    //Given
+    const currentTime = new Date();
+    const playerProfileModuleCore = testPlayerProfileModule(currentTime);
+    const firstName = 'Johnny';
+    const emailAddress = 'bravo@gmail.com';
+
+    const playerProfileWasCreated = new PlayerProfileWasCreated({
+      occurredAt: currentTime,
+      playerId: 'PlayerId',
+      firstName: firstName,
+      lastName: 'Bravo',
+      emailAddress: emailAddress,
+      phoneNumber: '123456789',
+    });
+
+    const commandPublisher = CommandPublisherMock(); // ?????
```

Comment:
@nowakprojects tutaj też nie wiem, czy to idzie puste? Chyba nie, to jakie parametry?

Reply:
Skąd moduł ma wiedzieć, że właśnie tego użyje jako commandPublishera :)? Popatrz na to co siedzi w test Module. Musisz jakoś podmienić obiekty na swoje mocki i potem zrobić expect na nich. Wszystko to nie bierze się z nikąd tylko jest spięte w kodzie. Ale test wygląda dobrze :)

Reply:
a dobra dobra teraz się zczaiłem, że przecież faktycznie commandPublisher to już idzie z modułu mail'i nie players-profile -.- eeee szkoda gadać... ale to nie zmienia faktu, że nie wiem w sumie jak dalej to sprawdzić :-P

Reply:
dobra z tego co widzę to musze zrobić też module core testowy do wysyłania maili, ale to jutro od rana, bo oczy mi się zamykają :-P

Reply:
Tak, bo nie jest to nigdzie zadeklarowane, o command potrzebuje miejsce na obsłużenie. 
Musiałbyś tak zrobić, żeby commandPublisher nie wysyłał rzeczywiście (w Twoim przypadku testowym) commanda, bo Ty tego nie potrzebujesz. Tobie tylko wystarczy, żeby nawet nic się nie stało - tylko po prostu sprawdzić czy metoda została wywołana.

Reply:
W tym przypadku metoda execute rzeczywiście próbuje wykonać command. A bez CommandHandlera nie ma jak go wykonać. Więc Tobie nie jest potrzebna rzeczywista implementacja akurat w tym teście (InMemoryComandBus), ale wystarczy Ci tylko zamockowana metoda execute i sprawdzenie czy była wywołana.

Reply:
`    let commandBus: CommandBus;
    commandBus = new InMemoryCommandBus();
    commandBus.execute = jest.fn();
`
jak to ma wyglądać ?- nie rozumiem tego dalej

Reply:
Co robi InMemoryCommandBus? Czy potrzebujesz czegokolwiek **w tym konkretnym teście** tutaj?

Reply:
1. no InMemoryCommandBus jest odpowiedzialny za metodę execute komend i za rejestrowanie ich handlerów za pomocą metody registerHandler
2. no wyżej pisałeś, że nie, ale, że potrzebuję zamockować metodę execute, to z jakiegoś miejsca to execute musi pochodzić, tak?